packages/playground/src/components/repo-settings/repo-branch-settings-rules/repo-branch-settings-rules-fields.tsx

```diff
@@ -0,0 +1,287 @@
+      <div className="flex gap-4">
+        <div className="flex-[2.5]">
+          <Input
+            id="target-patterns"
+            {...register!('targetPatterns')}
+            placeholder="Enter the target patterns"
+            autoFocus
+          />
+          <Text size={2} as="p" color="tertiaryBackground" className="max-w-[100%] mt-2">
+            Match branches using globstar patterns (e.g.”golden”, “feature-*”, “releases/**”)
+          </Text>{' '}
+        </div>
+        <Button
+          onClick={() => {}}
+          variant="split"
+          type="button"
+          className="pl-0 pr-0 min-w-28"
+          dropdown={
+            <DropdownMenu key="dropdown-menu">
+              <span>
+                <DropdownMenuTrigger insideSplitButton>
+                  <Icon name="chevron-down" className="chevron-down" />
+                </DropdownMenuTrigger>
+              </span>
+              <DropdownMenuContent align="end" className="mt-1">
+                <DropdownMenuGroup>
+                  <DropdownMenuItem onSelect={() => setSelectedOption('Include')}>Include</DropdownMenuItem>
+                  <DropdownMenuItem onSelect={() => setSelectedOption('Exclude')}>Exclude</DropdownMenuItem>
+                </DropdownMenuGroup>
+              </DropdownMenuContent>
+            </DropdownMenu>
+          }>
+          {selectedOption}
+        </Button>
+
+        {errors!.targetPatterns && (
+          <FormFieldSet.Message theme={MessageTheme.ERROR}>
+            {errors!.targetPatterns.message?.toString()}
+          </FormFieldSet.Message>
+        )}
+      </div>
+    </FormFieldSet.ControlGroup>
+  )
+}
+
+export const BranchSettingsRuleDefaultBranchField: React.FC<FieldProps> = ({ register, errors, watch, setValue }) => (
+  <FormFieldSet.ControlGroup className="min-h-8">
+    <FormFieldSet.Option
+      control={
+        <Checkbox
+          {...register!('defaultBranchValue')}
+          checked={watch!('defaultBranchValue')}
+          onCheckedChange={() => setValue!('defaultBranchValue', !watch!('defaultBranchValue'))}
+          id="default-branch"
+        />
+      }
+      id="default-branch"
+      label="Default Branch"
+    />
+
+    {errors!.defaultBranchValue && (
+      <FormFieldSet.Message theme={FormFieldSet.MessageTheme.ERROR}>
+        {errors!.defaultBranchValue.message?.toString()}
+      </FormFieldSet.Message>
+    )}
+  </FormFieldSet.ControlGroup>
+)
+
+export const BranchSettingsRuleBypassListField: React.FC<FieldProps> = ({ watch, setValue }) => (
+  <FormFieldSet.ControlGroup>
+    <FormFieldSet.Label htmlFor="bypassValue">Bypass list</FormFieldSet.Label>
+    <Select
+      value={watch!('bypassValue')}
+      onValueChange={value => setValue!('bypassValue', value, { shouldValidate: true })}>
+      <SelectTrigger id="bypassValue">
+        <SelectValue placeholder="Select users" />
+      </SelectTrigger>
+      <SelectContent>
+        <SelectItem value="1">user 1</SelectItem>
```

Comment:
why do we have hardcoded data in the reusable component? this data should be passed in via a prop